sunbeam-python/sunbeam/provider/maas/client.py

```diff
@@ -239,6 +239,18 @@ def list_machines_by_zone(client: MaasClient) -> dict[str, list[dict]]:
     return _group_machines_by_zone(machines_raw)
 
 
+def list_machines_by_matching_roles(client: MaasClient, roles: list[str]) -> list:
+    """List machines with given roles."""
+    filtered_machines = []
+    machines = list_machines(client)
```

Comment:
You should be able to do `list_machines(client, tags="role1,role2")` (I'm just not sure if tags takes a list or a comma separated string)

Reply:
`list_machines(client, tags="role1,role2")` worked and used it in subsequent commit